editor/src/messages/tool/tool_messages/spline_tool.rs

```diff
@@ -534,3 +534,308 @@ fn delete_preview(tool_data: &mut SplineToolData, responses: &mut VecDeque<Messa
 	tool_data.preview_point = None;
 	tool_data.preview_segment = None;
 }
+
+#[cfg(test)]
+mod test_spline_tool {
+	use crate::messages::input_mapper::utility_types::input_mouse::{EditorMouseState, ScrollDelta};
+	use crate::test_utils::test_prelude::*;
+
+	// Helper function to get spline points after drawing
+	async fn get_spline_points(editor: &mut EditorTestUtils) -> Option<Vec<DVec2>> {
+		let document = editor.active_document();
+		let network_interface = &document.network_interface;
+		let layer = network_interface.selected_nodes().selected_visible_and_unlocked_layers(network_interface).next()?;
+		println!("Found layer: {:?}", layer);
+		let vector_data = network_interface.compute_modified_vector(layer)?;
+		let points: Vec<DVec2> = vector_data.point_domain.positions().iter().cloned().collect();
+		println!("Found points: {:?}", points);
+		Some(points)
+	}
+
+	#[tokio::test]
+	async fn test_spline_with_zoomed_view() {
+		let mut editor = EditorTestUtils::create();
+		editor.new_document().await;
+
+		// Zooming the viewport
+		editor.handle_message(NavigationMessage::CanvasZoomSet { zoom_factor: 2.0 }).await;
+
+		// Selecting the spline tool
+		editor.select_tool(ToolType::Spline).await;
+
+		// Initiate drawing a spline
+		editor.left_mousedown(50.0, 50.0, ModifierKeys::empty()).await;
+		editor
+			.mouseup(
+				EditorMouseState {
+					editor_position: DVec2::new(50.0, 50.0),
+					mouse_keys: MouseKeys::empty(),
+					scroll_delta: ScrollDelta::default(),
+				},
+				ModifierKeys::empty(),
+			)
+			.await;
+
+		// Adding the second point
+		editor.left_mousedown(100.0, 50.0, ModifierKeys::empty()).await;
+		editor
+			.mouseup(
+				EditorMouseState {
+					editor_position: DVec2::new(100.0, 50.0),
+					mouse_keys: MouseKeys::empty(),
+					scroll_delta: ScrollDelta::default(),
+				},
+				ModifierKeys::empty(),
+			)
+			.await;
```

Comment:
Consider using `editor.click_tool(…)` to simplify the logic. You are also free to write your own helper functions e.g. on `EditorTestUtils`.